### ISSUE TITLE
Can't compile syn dependency due to missing feature

### ISSUE THREAD

**Issue by ExpHP** [whatisaphone/genawaiter#12]
```toml
[package]
name = "gen-example"
version = "0.1.0"
authors = ["Michael Lamparski <diagonaldevice@gmail.com>"]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
genawaiter = "0.99.0"
```

```
C:/Users/diago/.cargo/bin/cargo.exe run --color=always
   Compiling genawaiter-proc-macro v0.99.0
error[E0432]: unresolved imports `syn::FnArg`, `syn::ItemFn`
  --> C:\Users\diago\.cargo\registry\src\github.com-1ecc6299db9ec823\genawaiter-proc-macro-0.99.0\src\lib.rs:20:5
   |
20 |     FnArg,
   |     ^^^^^ no `FnArg` in the root
21 |     Ident,
22 |     ItemFn,
   |     ^^^^^^ no `ItemFn` in the root
```

It seems to me that `genawaiter-proc-macro` requires the `"full"` feature of `syn`, but does not enable it.  (How is this not failing on CI testing?)

**Comment by DevinR528** [comment]
Hmm that's a good question. When running the tests the features are included from the `[dev-dependencies]` right? My guess is that `futures-macro` uses `syn = ... features = ["full"]` so as features are additive this was always secretly enabled. I'll PR to add this.

Thanks for the catch!!

**Comment by whatisaphone** [comment]
Thanks for the report. Devin's fix has been released in v0.99.1.
